app-starters-common-analytics/src/main/java/org/springframework/cloud/stream/app/metrics/AggregateCounterReader.java

```diff
@@ -59,4 +61,9 @@
 	 * @return an object containing an indexed array of the counts .
 	 */
 	AggregateCounter getCounts(String name, int nCounts, DateTime end, AggregateCounterResolution resolution);
+
+	/**
+	 * List the names of all available aggregate counters.
+	 */
+	Collection<String> list();
```

Comment:
If we're gonna have something rather abstract, then I think Iterable is better, as it clearly states that this can't be modified. If we're gonna go the other way, then List is better IMO. I'd vote for the former